ui/lib/config-ui/addon/components/messages/page/create-and-edit-message-form.hbs

```diff
@@ -76,4 +83,29 @@
       />
     </Hds::ButtonSet>
   </div>
-</form>
\ No newline at end of file
+</form>
+
+{{#if this.showMessagePreviewModal}}
+  {{#if (eq @message.type "modal")}}
+    <Hds::Modal
+      id="message-modal-preview"
+      @size="large"
+      @color="warning"
+      @onClose={{fn (mut this.showMessagePreviewModal) false}}
+      data-test-modal="preview modal"
+      as |M|
+    >
+      <M.Header data-test-modal-title={{@message.title}}>
+        {{@message.title}}
+      </M.Header>
+      <M.Body data-test-modal-body>
+        {{@message.message}}
+      </M.Body>
+      <M.Footer as |F|>
+        <Hds::Button @text="Confirm" {{on "click" F.close}} data-test-modal-button="Close" />
+      </M.Footer>
+    </Hds::Modal>
+  {{else}}
+    <Messages::PreviewImage @message={{@message}} @showMessagePreviewModal={{this.showMessagePreviewModal}} />
```

Comment:
This component is basically just an HDS modal. I put it in a component since I needed to override some height and width styles for the modal.
<img width="1724" alt="Screenshot 2023-12-20 at 1 20 06 PM" src="https://github.com/hashicorp/vault/assets/30884335/406f9aff-f592-44a6-a254-ea7a9c52def6">